core/math/rect2.h

```diff
@@ -370,6 +369,16 @@ struct [[nodiscard]] Rect2 {
 			position(p_pos),
 			size(p_size) {
 	}
+	_FORCE_INLINE_ Rect2(const Rect2i &p_rect);
 };
 
+#ifdef RECT2I_H
+Rect2::Rect2(const Rect2i &p_rect) :
+		position(p_rect.position),
+		size(p_rect.size) {}
+Rect2i::Rect2i(const Rect2 &p_rect) :
+		position(p_rect.position),
+		size(p_rect.size) {}
+#endif // RECT2I_H
```

Comment:
I guess this wouldn't play nice with `#pragma once`.

I don't fully understand how this works but it seems to make things quite dependent of include order?